<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
  <!-- Arguments -->
  <arg name="hardware_version" default="ned"/>

  <param name="robot_description" command="$(find xacro)/xacro $(find niryo_robot_description)/urdf/ned/niryo_ned.urdf.xacro"/>
    
  <group ns="niryo_robot_hardware_interface" >
    <rosparam file="$(find ttl_driver)/config/$(arg hardware_version)/bus_config.yaml" command="load" ns="ttl_driver"/>
    <rosparam file="$(find ttl_driver)/config/$(arg hardware_version)/motors_config.yaml" command="load" ns="ttl_driver"/>
    <rosparam file="$(find ttl_driver)/config/$(arg hardware_version)/dynamixels_params.yaml" command="load" />
    <rosparam file="$(find ttl_driver)/config/default.yaml" command="load"  ns="ttl_driver"/>

    <rosparam file="$(find can_driver)/config/bus_config.yaml" command="load" ns="can_driver"/>
    <rosparam file="$(find can_driver)/config/motors_config.yaml" command="load" ns="can_driver"/>
    <rosparam file="$(find can_driver)/config/steppers_params.yaml" command="load" />
    <rosparam file="$(find can_driver)/config/default.yaml" command="load" ns="can_driver"/>
  
    <node pkg="joints_interface" type="joints_interface_node" name="joints_interface" output="screen">
      <rosparam file="$(find joints_interface)/config/$(arg hardware_version)/joints_params.yaml" command="load" />
      <rosparam file="$(find joints_interface)/config/default.yaml" command="load" />
    </node>
    
  </group>

  <!-- spawn controllers -->
  <rosparam command="load" file="$(find joints_interface)/config/niryo_ned_controllers.yaml" />

  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="log"
      args="joint_state_controller niryo_robot_follow_joint_trajectory_controller --shutdown-timeout 1"/>

  <!-- robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" />  
  
</launch>
